base/src/main/java/io/spine/base/Environment.java

```diff
@@ -69,101 +103,55 @@ public Environment createCopy() {
     }
 
     /**
-     * Restores the state from the instance created by {@link #createCopy()}.
+     * Determines the current environment type.
      *
-     * <p>Call this method when cleaning up tests that modify {@code Environment}.
-     */
-    @VisibleForTesting
-    public void restoreFrom(Environment copy) {
-        // Make sure this matches the set of fields copied in the copy constructor.
-        this.tests = copy.tests;
-    }
-
-    /**
-     * Verifies if the code currently runs under a unit testing framework.
+     * <p>If {@linkplain #registerCustom(EnvironmentType) custom env types have been defined},
+     * goes through them in an undefined order. Then, checks the {@linkplain BaseEnvironmentType
```

Comment:
I don't think the order is undefined. Let's check those registered by the user in the order of their registration. I.e. if I register some `STAGING` first, and `PRE_PRODUCTION` second, we should check `STAGING` first.

Reply:
It's actually defined, the last environment to `register` is the first one that checked.

I thought about the following scenario:

Having two env types that are not mutually exclusive is a bug, but if the env types are checked in a predictable order, the bug becomes harder to discover.